terminate called after throwing an instance of 'std::invalid_argument'
  what():  statements for distribution must be in the same level-1 subloop
Aborted (core dumped)
